### PR TITLE
[Loupe Finding] Prevent maker reboot recovery from discarding funded swapcoins

### LINKED ISSUE #878: [Loupe Finding] bug: Maker reboot recovery can discard funded unfinished swapcoins when tracker state is missing
## Description
Maker startup recovery currently scans the wallet for `unfinished incoming/outgoing` swapcoins and starts `recover_from_swap` if any are found. However, on reboot it may use a synthetic swap id like `reboot-recovery-<timestamp>` instead of the original persisted swap id.

`recover_from_swap` then checks `maker_swap_tracker.cbor` for that synthetic id:
```rust
get_record(&swap_id)
    .map(|r| r.funding_broadcast)
    .unwrap_or(false)
```
If no tracker record exists, the code defaults `funding_broadcast` to `false` and concludes that no funding was broadcast. It then removes the persisted incoming and outgoing swapcoins from the maker wallet.

## Steps to Reproduce
Create a test that 
1.Runs a 2-maker Taproot swap.
2.Uses CloseAtPrivateKeyHandover for Maker2 so Maker2 has persisted unfinished swapcoins.
3.Restarts Maker2 before idle recovery writes a tracker record.
4.Verifies Maker2 still has unfinished incoming/outgoing swapcoins after reboot recovery.

## Expected Behavior
The maker should recover it's swapcoin as expected.

## Actual Behavior
Observed in testing locally
```rust
2026-05-25T00:30:58.590914+05:30 INFO coinswap::wallet::api - Unfinished swaps - Incoming: 1, Outgoing: 1
2026-05-25T00:30:58.590934+05:30 INFO coinswap::maker::server - [12833] Incomplete swaps detected on startup: 1 incoming, 1 outgoing. Starting recovery.
2026-05-25T00:30:58.591029+05:30 INFO coinswap::maker::server - [12833] Bitcoin Network: regtest
2026-05-25T00:30:58.591070+05:30 INFO coinswap::maker::server - [12833] Spendable Wallet Balance: 0.14501458 BTC
2026-05-25T00:30:58.591089+05:30 INFO coinswap::maker::server - [12833] Server setup complete! Listening on port 12833
2026-05-25T00:30:58.591515+05:30 INFO coinswap::maker::server - [12833] recover_from_swap started | height=235 timelock_expiry=305 | incoming=1 outgoing=1
2026-05-25T00:30:58.591540+05:30 INFO coinswap::maker::server - [12833] Funding was never broadcast for swap reboot-recovery-1779649258 — nothing to recover. Discarding swapcoins.
2026-05-25T00:30:58.591686+05:30 INFO coinswap::maker::rpc::server - [7415] RPC socket binding successful at 127.0.0.1:7415
2026-05-25T00:30:58.591698+05:30 INFO coinswap::wallet::api - Removed outgoing swapcoin: 23ce6e2d304f1aad2ac00811463867b469fbd434aa79487b83b574f7e7765a27 (remaining: 0)
2026-05-25T00:30:58.591745+05:30 INFO coinswap::wallet::api - Removed incoming swapcoin from wallet store: 316536b8df01106db25ea613508208bfbedac42845196ed931374ffecfc09b90 (remaining: 0)
```

### PR DESCRIPTION
**Pull request by 0xEgao**
# Pull Request

## Description
Fixes #878.
Only use persisted swapcoin swap IDs during maker startup recovery and only discard unfinished swapcoins when tracker state explicitly says funding was not broadcast.

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [x] Both
- [ ] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [x] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [x] All unit tests pass (`cargo test`)
- [x] Integration tests pass (`cargo test --features integration-test`)
- [x] Changes were manually tested on **regtest**
- [x] End-to-end maker ↔ taker swap tested (where applicable)
- [x] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [x] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## Summary by CodeRabbit

* **Bug Fixes**
  * Reboot recovery now groups unfinished swaps by their persisted identifier and performs recovery once per group, preserving swap identity when available.
  * If funding broadcast status is explicitly recorded as not broadcast, recovery discards wallet state; otherwise it continues recovery/monitoring.
  * Startup recovery now fails instead of generating a fallback identifier when persisted unfinished items lack an identifier.

* **Tests**
  * Added an integration test covering Taproot reboot recovery to confirm funded incoming/outgoing unfinished swaps are preserved across a maker restart.

### CODE REVIEW

File: src/maker/server.rs [hunk 1]
```diff
@@ -91,6 +91,12 @@ pub fn start_server(maker: Arc<MakerServer>) -> Result<(), MakerError> {
                 out.len()
             );
 
+            // QA: Reboot recovery could discard funded swapcoins after treating
+            // the swap as "funding was never broadcast". Group unfinished coins
+            // by swap id before recovery so each discard/recovery decision is
+            // scoped to the swap being evaluated, preserving funded recovery
+            // material across restarts.
+            // Regression coverage: `tests/integration/taproot_reboot_recovery.rs`.
```
Comment (0xEgao) [inline src/maker/server.rs:99]: Added a sample QA work comment, if this looks good, i will add this for all the past QA work merged in this pr itself.
cc: @mojoX911

File: src/maker/server.rs [hunk 2]
```diff
@@ -91,17 +91,43 @@ pub fn start_server(maker: Arc<MakerServer>) -> Result<(), MakerError> {
...
+                groups
+                    .entry(outgoing.swap_id.clone())
+                    .or_insert_with(|| (Vec::new(), Vec::new()))
+                    .1
+                    .push(outgoing);
+            }
+
+            for (swap_id, (inc, out)) in groups {
+                let swap_id = swap_id.unwrap_or_else(|| {
+                    format!("reboot-recovery-{}", super::swap_tracker::now_secs())
```
Comment (mojoX911) [inline src/maker/server.rs]: This seems like a deep one. 

My first question is, why is there this `reboot_recovery` random string used as swap_id? And why should swap_id ever be Optional?

There should be no situation in which the swap_id of an incoming/outgoing swapcoin is None. So maybe we can tackle that problem at the root and remove this random string approach altogether.

File: src/maker/server.rs (hunk 1 above)
Comment (mojoX911) [reply src/maker/server.rs:99]: Yes looks good..

File: src/maker/server.rs (hunk 2 above)
Comment (0xEgao) [reply src/maker/server.rs]: I have removed this synthetic `reboot-recovery-{timestamp}` swap id creation at all, for converting the `swap_id` to a `String` from an `Option<String>` will required a large amount of changeset. 

I think we can do it in a seperate pr, what do you say?

**Review by stark-3k — APPROVED** [review]
Ack

**Review by mojoX911 — APPROVED** [review]
Ack
